### ISSUE TITLE
SNOW-887604: 3.1.0a1: Connection initiation never completes due to loop on ocsp_response_validation_cache.lock

### ISSUE THREAD

**Issue by jhowa1** [snowflakedb/snowflake-connector-python#1683]
### Python version

Python 3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)]

### Operating system and processor architecture

Windows-10-10.0.22621-SP0

### Installed packages

```python
alabaster==0.7.13
altgraph==0.17.3
asn1crypto==1.5.1
black==23.3.0
boto3==1.20.35
botocore==1.23.54
cachetools==5.3.1
certifi==2022.12.7
cffi==1.15.1
chardet==5.1.0
charset-normalizer==2.1.1
click==8.1.3
colorama==0.4.6
connectorx==0.3.1
cryptography==40.0.1
dill==0.3.6
et-xmlfile==1.1.0
filelock==3.10.7
flatsplode==0.5.0
greenlet==2.0.2
grpcio==1.56.0
grpcio-status==1.56.0
gzip-reader==0.1
httplib2==0.22.0
idna==3.4
imagesize==1.4.1
importlib-metadata==6.8.0
isodate==0.6.1
isort==5.12.0
Jinja2==3.1.2
jmespath==0.10.0
jsonpath-ng==1.5.3
lazy-object-proxy==1.9.0
lxml==4.9.2
MarkupSafe==2.1.2
mccabe==0.7.0
mypy-extensions==1.0.0
numpy==1.24.2
oauth2client==4.1.3
oauthlib==3.2.2
openpyxl==3.1.2
opentelemetry-api==1.19.0
opentelemetry-instrumentation==0.40b0
oscrypto==1.3.0
packaging==23.0
pathspec==0.11.1
pefile==2023.2.7
Pillow==9.5.0
platformdirs==3.2.0
ply==3.11
proto-plus==1.22.3
protobuf==4.22.1
pyasn1==0.5.0
pyasn1-modules==0.3.0
pycparser==2.21
pycryptodomex==3.17
pydata-google-auth==1.8.1
Pygments==2.15.1
pyinstaller==5.12.0
pyinstaller-hooks-contrib==2023.3
PyJWT==2.6.0
pylint==2.17.4
pyOpenSSL==23.1.1
pyorc==0.8.0
pyparsing==3.1.0
python-dateutil==2.8.2
python-tds==1.11.0
pytz==2023.3
pytz-deprecation-shim==0.1.0.post0
pywin32==305
pywin32-ctypes==0.2.0
regex==2023.3.23
requests==2.28.2
requests-file==1.5.1
requests-oauthlib==1.3.1
requests-toolbelt==1.0.0
rsa==4.9
s3transfer==0.5.2
simple-salesforce==1.12.4
six==1.16.0
snowballstemmer==2.2.0
snowflake-connector-python==3.1.0a1
tomli==2.0.1
tomlkit==0.11.8
Trac==1.4.3
types-pytz==2023.3.0.0
typing_extensions==4.5.0
tzdata==2023.3
tzlocal==4.3
uritemplate==4.1.1
urllib3==1.26.15
wrapt==1.15.0
xmltodict==0.13.0
zeep==4.2.1
zipp==3.16.2
```

### What did you do?

```python
This connection has been in use since September, 2022, unchanged.

cnxn = Connector.connect(
        user= ,
        password= ,
        account= ,
        database= ,
        warehouse=,
        schema=,
        role= ,
    )
```

### What did you expect to see?

A connection.

### Can you set logging to DEBUG and collect the logs?

```bash
2023-08-05 19:05:22,396 - INFO - connection.py:313 - Snowflake Connector for Python Version: 3.1.0a1, Python Version: 3.10.7, Platform: Windows-10-10.0.22621-SP0
2023-08-05 19:05:22,397 - DEBUG - connection.py:600 - connect
2023-08-05 19:05:22,397 - DEBUG - connection.py:918 - __config
2023-08-05 19:05:22,397 - INFO - connection.py:1055 - This connection is in OCSP Fail Open Mode. TLS Certificates would be checked for validity and revocation status. Any other Certificate Revocation related exceptions or OCSP Responder failures would be disregarded in favor of connectivity.
2023-08-05 19:05:22,397 - INFO - connection.py:1073 - Setting use_openssl_only mode to False
2023-08-05 19:05:22,398 - DEBUG - converter.py:146 - use_numpy: False
2023-08-05 19:05:22,398 - DEBUG - converter_issue23517.py:27 - initialized
2023-08-05 19:05:22,398 - DEBUG - connection.py:798 - REST API object was created: [mySF_ACCOUNT].snowflakecomputing.com:443
2023-08-05 19:05:22,398 - DEBUG - _auth.py:171 - authenticate
2023-08-05 19:05:22,398 - DEBUG - _auth.py:201 - assertion content: *********
2023-08-05 19:05:22,398 - DEBUG - _auth.py:204 - account=[mySF_ACCOUNT], request_id=f2f89dc0-b1a1-41f6-8be2-13e1c60307d6
2023-08-05 19:05:22,398 - DEBUG - _auth.py:237 - body['data']: {'CLIENT_APP_ID': 'PythonConnector', 'CLIENT_APP_VERSION': '3.1.0', 'SVN_REVISION': None, 'ACCOUNT_NAME': '[mySF_ACCOUNT]', 'LOGIN_NAME': '[user]', 'CLIENT_ENVIRONMENT': {'APPLICATION': 'PythonConnector', 'OS': 'Windows', 'OS_VERSION': 'Windows-10-10.0.22621-SP0', 'PYTHON_VERSION': '3.10.7', 'PYTHON_RUNTIME': 'CPython', 'PYTHON_COMPILER': 'MSC v.1933 64 bit (AMD64)', 'OCSP_MODE': 'FAIL_OPEN', 'TRACING': 10, 'LOGIN_TIMEOUT': 120, 'NETWORK_TIMEOUT': None}, 'SESSION_PARAMETERS': {'CLIENT_PREFETCH_THREADS': 4}}
2023-08-05 19:05:22,399 - DEBUG - _auth.py:247 - Timeout set to 120
2023-08-05 19:05:22,399 - DEBUG - retry.py:351 - Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
2023-08-05 19:05:22,399 - DEBUG - network.py:1207 - Session status for SessionPool '[mySF_ACCOUNT].snowflakecomputing.com', SessionPool 1/1 active sessions
2023-08-05 19:05:22,399 - DEBUG - network.py:873 - remaining request timeout: 120, retry cnt: 1
2023-08-05 19:05:22,400 - DEBUG - network.py:855 - Request guid: b8b26541-f201-4edc-9dc1-0adaef5735d0
2023-08-05 19:05:22,400 - DEBUG - network.py:1063 - socket timeout: 60
2023-08-05 19:05:22,403 - DEBUG - connectionpool.py:1003 - Starting new HTTPS connection (1): [mySF_ACCOUNT].snowflakecomputing.com:443
2023-08-05 19:05:23,195 - DEBUG - ssl_wrap_socket.py:79 - OCSP Mode: FAIL_OPEN, OCSP response cache file name: None
2023-08-05 19:05:23,608 - DEBUG - _api.py:181 - Attempting to acquire lock 1352864771104 on C:\Users\Administrator\AppData\Local\Snowflake\Caches\ocsp_response_validation_cache.lock
2023-08-05 19:05:23,609 - DEBUG - _api.py:184 - Lock 1352864771104 acquired on C:\Users\Administrator\AppData\Local\Snowflake\Caches\ocsp_response_validation_cache.lock
2023-08-05 19:05:23,614 - DEBUG - _api.py:216 - Attempting to release lock 1352864771104 on C:\Users\Administrator\AppData\Local\Snowflake\Caches\ocsp_response_validation_cache.lock
2023-08-05 19:05:23,615 - DEBUG - _api.py:219 - Lock 1352864771104 released on C:\Users\Administrator\AppData\Local\Snowflake\Caches\ocsp_response_validation_cache.lock
```

**Comment by jhowa1** [comment]
To clarify the loop - the loop starts at _api.py:181 through :219.   This four-step loop repeats until python.exe is killed.    The full debug log records 142,000 iterations of this loop in the space of 20 minutes.

**Comment by sfc-gh-sfan** [comment]
3.1.0a1 only adds nanoarrow, which should not impact the behavior of connection. Do you happen to know the previous version that works for you, and have you tried 3.1.0?

cc @sfc-gh-aling @sfc-gh-mkeller for potentially OCSP cache issue.

**Comment by sfc-gh-aling** [comment]
hey @jhowa1 , we've identified the issue in the version 3.1.0a1 and fixed it in 3.1.0a2.
please try version 3.1.0a2 and let us know how it works for you

**Comment by jhowa1** [comment]
Installed and test processes ran uneventfully.  Thank you for the quick
response!
